brain/synapse.cpp

```diff
@@ -0,0 +1,175 @@
+/* Copyright (c) 2013-2016, EPFL/Blue Brain Project
+ *                          Daniel.Nachbaur@epfl.ch
+ *
+ * This file is part of Brion <https://github.com/BlueBrain/Brion>
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License version 3.0 as published
+ * by the Free Software Foundation.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License for more
+ * details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+
+
+#include "synapse.h"
+#include "synapses.h"
+
+#include <vmmlib/vector.hpp>
+
+namespace brain
+{
+
+struct Synapse::Impl
+{
+    Impl( const Synapses& synapses, const size_t index )
+        : _synapses( synapses )
+        , _index( index )
+    {}
+
+    const Synapses& _synapses;
```

Comment:
This is a bit unconvenient for the wrapping, I'll have to think about how to deal with memory management. Since probably I'll have to introduce some type of shared pointer I think it's preferable to not use pimpl here to reduce the amount of heap objects that are needed.